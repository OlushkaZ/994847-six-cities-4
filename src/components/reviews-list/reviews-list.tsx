import * as React from 'react';
import {connect} from "react-redux";

import {Review} from '../../types/review';
import {getSortedReviewsSlice} from '../../utils';
import ReviewCard from '../review-card/review-card';
import CommentSubmissionForm from '../comment-submission-form/comment-submission-form';

interface Props {
  reviews: Review[];
  authorizationStatus: string;
}

const ReviewsList: React.FC<Props> = ({reviews, authorizationStatus}) => {
  const sortedReviews = getSortedReviewsSlice(reviews);

  return (
    <section className="property__reviews reviews">
      <h2 className="reviews__title">
        Review{reviews.length !== 1 ? `s` : ``} &middot; <span className="reviews__amount">{reviews.length}</span>
      </h2>
      <ul className="reviews__list">
        {sortedReviews.map((review, index) => (
          <ReviewCard key={index} review={review}/>
        ))}
      </ul>
      {(authorizationStatus === `AUTH`)
        ? (
          <CommentSubmissionForm />
        ) : null
      }
    </section>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: state.user.authorizationStatus,
  reviews: state.reviews.reviews,
});

export {ReviewsList};
export default connect(mapStateToProps)(ReviewsList);


