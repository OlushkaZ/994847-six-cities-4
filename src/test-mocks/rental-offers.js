export const RENTAL_OFFERS = [
  {
    id: 1,
    location: {
      cityCoordinates: [52.3909553943508, 4.85309666406198],
    },
    img: `img/apartment-01.jpg`,
    price: 100,
    rating: `95%`,
    name: `Beautiful luxurious apartment at great location`,
    type: `hostel`,
    isPremium: true,
    isBookmark: false,
  },

  {
    id: 2,
    location: {
      cityCoordinates: [52.3909553943508, 4.85309666406198],
    },
    img: `img/apartment-02.jpg`,
    price: 193,
    rating: `60%`,
    name: `Canal View Prinsengracht`,
    type: `apartment`,
    isPremium: true,
    isBookmark: true,
  },

  {
    id: 3,
    location: {
      cityCoordinates: [52.3909553943508, 4.85309666406198],
    },
    img: `img/apartment-03.jpg`,
    price: 250,
    rating: `16%`,
    name: `Nice, cozy, warm big bed apartment`,
    type: `hotel`,
    isPremium: false,
    isBookmark: false,
  },

  {
    id: 4,
    location: {
      cityCoordinates: [52.3909553943508, 4.85309666406198],
    },
    img: `img/apartment-01.jpg`,
    price: 50,
    rating: `75%`,
    name: `Wood and stone place`,
    type: `house`,
    isPremium: false,
    isBookmark: true
  }
];

export const RENTAL_OFFER = {
  id: 1,
  img: `img/apartment-01.jpg`,
  price: 100,
  rating: `95%`,
  name: `Beautiful luxurious apartment at great location`,
  type: `hostel`,
  isPremium: true,
  isBookmark: false
};

