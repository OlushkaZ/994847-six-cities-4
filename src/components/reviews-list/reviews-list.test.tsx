import React from 'react';
import {createStore} from 'redux';
import {Provider} from "react-redux";
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import reducer from '../../reducer/combine-reducer';
import {REVIEWS} from '../../test-mocks/reviews';
import {ReviewsList} from './reviews-list';

const store = createStore(reducer);

it(`Should render ReviewsList correctly`, () => {
  const tree = renderer.create((
    <Provider store={store}>
      <BrowserRouter>
        <ReviewsList reviews={REVIEWS} authorizationStatus={`AUTH`} />
      </BrowserRouter>
    </Provider>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});

