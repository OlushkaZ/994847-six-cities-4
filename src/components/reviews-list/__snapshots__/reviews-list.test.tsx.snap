// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ReviewsList correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Review
    s
     Â· 
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="/img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Gary
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": 2,
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Left my sweater in the hotel. Mr. Herbert, sent home the sweater for free. Great service.
        </p>
        <time
          className="reviews__time"
          dateTime="2008-02-01"
        >
          February 2008
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="/img/avatar.svg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Michael
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": 3,
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The proximity to the airport is great, very friendly owner, clean and tidy, parking right outside the door - a very good choice with an early onward journey to the airport with excellent value for money.
        </p>
        <time
          className="reviews__time"
          dateTime="2007-05-01"
        >
          May 2007
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="/img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Carrie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": 5,
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We felt very welcome as guests.The room, located on the noisy thoroughfare, was absolutely soundproofed.We slept very well in the beds.Parking in front of the hotel, perfect!
        </p>
        <time
          className="reviews__time"
          dateTime="2003-04-01"
        >
          April 2003
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-stars"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value=""
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
